plugins/inputs/interrupts/interrupts.go

```diff
@@ -115,13 +176,14 @@ func (s *Interrupts) Gather(acc telegraf.Accumulator) error {
 			acc.AddError(fmt.Errorf("Could not open file: %s", file))
 			continue
 		}
-		defer f.Close()
 		irqs, err := parseInterrupts(f)
+		_ = f.Close()
```

Comment:
Please don't do this. This reduces the code-robustness just to make the linter happy. Disable the linter error for `defer f.Close()` with a comment saying that we cannot do anything if `Close()` fails thus not checking the return code.

Reply:
However, you're advocating defer inside a loop, which checkers complain about because it could leak. I haven't measured a leak with the prior behaviour, but I only test on a limited set of golang compilers and OSes. With this change, there is no chance of leak either way.

Reply:
You are right... So the "correct" solution would be to put the reading code into a separate function
```golang
func readInterrupts(file) ([]IRQ, error)
	f, err := os.Open(file)
	if err != nil {
		return nil, fmt.Errorf("could not open file: %s", file)
	}
	defer f.Close()
	return parseInterrupts(f)
}
```

Reply:
When moving the spurious counters into the fields, it occurs to me that it could be refactored to be more clear. To wit, we don't have spurious counters for softirqs. I'll try some things and see how it looks.